src/main/scala/firrtl/transforms/ConstantPropagation.scala

```diff
@@ -378,6 +378,8 @@ class ConstantPropagation extends Transform {
             nodeMap(lname) = constPropExpression(nodeMap, instMap, constSubOutputs)(pad(fval, ltpe))
           case Mux(_, tval: Literal, fval: WRef, _) if weq(lref, fval) =>
             nodeMap(lname) = constPropExpression(nodeMap, instMap, constSubOutputs)(pad(tval, ltpe))
+          case WRef(`lname`, _,_,_) => // If a register is connected to itself, propagate zero
```

Comment:
Why are the back ticks here?

Reply:
Backticks mean check if the value of the field is equal to the variable in backticks rather than binding a new variable name, ie.
```scala
case WRef(`lname`, _,_,_) => // note lname is a variable name bound above
// Is the same as:
case WRef(rname, _,_,_) if lname == rname =>
```

Reply:
Thanks, I just looked it up.  I've never used that feature.

Reply:
I discovered it a few months ago but haven't had many opportunities to use it yet! I'm a big fan.